packages/nx-monorepo/src/nx-monorepo.ts

```diff
@@ -27,6 +27,23 @@ import { DEFAULT_CONFIG, SyncpackConfig } from "./syncpack-options";
 
 const NX_MONOREPO_PLUGIN_PATH: string = ".nx/plugins/nx-monorepo-plugin.js";
 
+/**
+ * Execute command to run based on package manager configured.
+ *
+ * @param packageManager package manager being used.
+ */
+export function execute(packageManager: NodePackageManager) {
+  switch (packageManager) {
+    case NodePackageManager.YARN:
+    case NodePackageManager.YARN2:
+      return "yarn";
+    case NodePackageManager.PNPM:
+      return "pnpx";
+    default:
+      return "npx";
+  }
+}
+
```

Comment:
`execute` is a verb and expect this function to actually perform the execution, and would expect to be able to pass the command to execute as a parameter as well - consider renaming.

Would also be nice to have this take optional command that would get appended to clean up the implementations below: `${execute(this.package.packageManager)} foobar` to `execute(this.package.packageManager, "foobar")` (both of which result in `pnpx foobar`, etc.

Maybe we should add a public method on `NxMonorepo` for this, which would already have context of `this.package.packageManager` and support subclasses or child project to utilize this functionality.

```ts
class NxMonorepo {
 //...
 buildExecutableCommand(...command: string[]): string {
   return execute(this.package.packageManager) + " " + command.join(" ");
 }
}
```